stl/src/syncstream.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// initialize syncstream mutex map
+
+#include <internal_shared.h>
+#include <map>

```

Comment:
`map` is an STL container whose representation is affected by `_ITERATOR_DEBUG_LEVEL`. This means that we can't drag it in to the import lib, due to ODR violations. (There are only 2 DLLs/import LIBs for release/debug, yet there are 5 possible settings for IDL.) This is the scenario that the Scary Comment For Import Lib Stuff warns against (with `basic_string` as the example).

If this code is moved into a satellite DLL (as I suspect is necessary), then this issue is moot - satellite DLLs can use STL containers without fear of ODR violations because there is a binary boundary.